### PR TITLE
Elixir 1.14 compatibility

### PR DESCRIPTION
**Pull request by marcandre**
Hi! I was curious to try this on our codebase but ran into issues as we're already using Elixir 1.14

This fixes the two incompatibilities with Elixir 1.14:

* `Code.Identifier.inspect_as_key` was removed in Elixir 1.14
* AST for imported functions was changed in Elixir 1.14

It also resolves an issue when running the tests by disabling async mode for the parser tests.

~It also includes #144 (typo in commit message fixed)~ Ah! Already merged 👍

### CODE REVIEW

**Comment by marcandre** [comment]
I bumped the github action too, but the tests don't seem to be running...

**Comment by Qqwy** [comment]
Very strange, let me investigate.

File: .github/workflows/ci.yml
Comment (Qqwy) [inline .github/workflows/ci.yml:16]: :+1:

File: lib/type_check/builtin/compound_fixed_map.ex
Comment (Qqwy) [inline lib/type_check/builtin/compound_fixed_map.ex:82]: This is a nice way to solve the problem! :green_heart:

File: test/type_check/internals/parser_test.exs
Comment (Qqwy) [inline test/type_check/internals/parser_test.exs:3]: Interesting. I'll do some checking afterwards to see whether we can go back to `async`, because there is no clear reason why it should not be possible.

File: lib/type_check/builtin/compound_fixed_map.ex
Comment (Qqwy) [inline lib/type_check/builtin/compound_fixed_map.ex:82]: This is a nice way to solve the problem! :green_heart:

**Review by Qqwy — APPROVED** [review]

File: lib/type_check/builtin/compound_fixed_map.ex
Comment (marcandre) [reply lib/type_check/builtin/compound_fixed_map.ex:82]: Actually it doesn't work... Looks like Elixir is checking the `import` in 1.14 even if it is not run. Reverted PR to use `defdelegate` which was my original version

File: test/type_check/internals/parser_test.exs
Comment (marcandre) [reply test/type_check/internals/parser_test.exs:3]: Yeah, I agree in general and I'm opening an issue in Elixir. In this case though, I'm not sure that defining the same module "simultaneously" is the way to go...

**Comment by Qqwy** [comment]
It seems like the workflow is not running as a security measure. [See this doc page from GitHub](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks). However, I cannot find the 'Approve and run' button that should be visible to start the workflow run manually.

File: test/type_check/internals/parser_test.exs
Comment (Qqwy) [reply test/type_check/internals/parser_test.exs:3]: I recommend waiting with that; I think that there is a 90% chance that the issue is in this library rather than in Elixir v1.14. :sweat_smile:

Comment (marcandre) [reply test/type_check/internals/parser_test.exs:3]: FYI, I opened https://github.com/elixir-lang/elixir/issues/12189

**Comment by marcandre** [comment]
> It seems like the workflow is not running as a security measure. [See this doc page from GitHub](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks). However, I cannot find the 'Approve and run' button that should be visible to start the workflow run manually.

Yeah, I was looking for that too.

Let me try removing the commit that changes the github action

File: test/type_check/internals/parser_test.exs
Comment (Qqwy) [reply test/type_check/internals/parser_test.exs:3]: You managed to extract a very nice concise counter-example, so maybe it is a problem there? let's see what the Elixir team thinks.

**Comment by marcandre** [comment]
Doesn't show the usual message "CI will run after approval from someone higher in the food chain for this project" or whatever usually shows 🤷‍♂️

**Comment by Qqwy** [comment]
:shrug: Then I assume that GitHub has problems right now. I will run the tests of this PR on Elixir v1.14 and v1.13 locally.

Please let me know if you're happy with the changes as they are now, or whether I should wait before merging.

**Comment by marcandre** [comment]
> Please let me know if you're happy with the changes as they are now, or whether I should wait before merging.

Ah ah, yeah, that will hopefully teach me not to get cute _after creating a PR_ and screwing things up.

So code should be good now, all green in 1.13.4 ad 1.14.1 (on my machine)

BTW, the sync code doesn't take long... "Finished in 37.4 seconds (34.4s async, 3.0s sync)"

**Comment by Qqwy** [comment]
> Ah ah, yeah, that will hopefully teach me not to get cute after creating a PR and screwing things up.

No worries! :stuck_out_tongue_winking_eye: 

Since a few months, GitHub has the possibility to switch PRs between 'draft' and 'ready' mode by the way.

**Comment by Qqwy** [comment]
Tests are also green here; merging

**Comment by marcandre** [comment]
@Qqwy Will you be making a release?

**Comment by Qqwy** [comment]
Yes!
I wanted to wait for the CI on master to finish first, just to be sure :wink: .

Version 0.12.3 has been released which contains this PR.

**Comment by marcandre** [comment]
Awesome, thanks @Qqwy ❤️

**Pull request merged**
